core/src/lib.rs

```diff
@@ -102,25 +102,27 @@ where
 }
 
 /// Create a worker for replaying a specific history. It will auto-shutdown as soon as the history
-/// has finished being replayed. The provided client should be a mock, and this should only be used
-/// for workflow testing purposes.
-pub fn init_replay_worker(
+/// has finished being replayed.
+pub fn init_replay_worker<I>(
     mut config: WorkerConfig,
-    history: &History,
-) -> Result<Worker, anyhow::Error> {
+    histories: I,
+) -> Result<Worker, anyhow::Error>
+where
+    I: IntoIterator<Item = History> + 'static,
```

Comment:
The sync iterator approach is not going to work well for TS / Python.
IMHO we need a way to asynchronously feed histories, the "feed" method should block if concurrently replays exceed a configurable threshold to avoid process memory growing unbounded.